flink-connectors/flink-connector-kafka/src/main/java/org/apache/flink/connector/kafka/source/reader/KafkaPartitionSplitReader.java

```diff
@@ -414,6 +436,45 @@ private void maybeRegisterKafkaConsumerMetrics(
         }
     }
 
+    private Supplier<Long> getBytesConsumedTotalSupplier(KafkaConsumer<?, ?> consumer) {
+        final String bytesConsumedTotalName = "bytes-consumed-total";
+        final Optional<Supplier<Long>> supplier =
+                consumer.metrics().entrySet().stream()
+                        .filter(
+                                e ->
+                                        e.getKey()
+                                                        .group()
+                                                        .equals(
+                                                                KAFKA_METRIC_GROUP_CONSUMER_FETCH_MANAGER_METRICS)
+                                                && e.getKey()
+                                                        .name()
+                                                        .equals(KAFKA_METRIC_BYTES_CONSUMED_TOTAL))
+                        .map(Map.Entry::getValue)
+                        .findFirst()
```

Comment:
I'd extract that to `Optional<Metric> findKafkaMetic(String group, String name)`, then you can reuse it for the lag metric.